pkg/handlers/traceflow/handler.go

```diff
@@ -0,0 +1,177 @@
+	logger    logr.Logger
+	k8sClient dynamic.Interface
+	clock     clock.Clock
+}
+
+func newRequestsHandlerWithClock(logger logr.Logger, k8sClient dynamic.Interface, clock clock.Clock) *requestsHandler {
+	return &requestsHandler{
+		logger:    logger,
+		k8sClient: k8sClient,
+		clock:     clock,
+	}
+}
+
+func NewRequestsHandler(logger logr.Logger, k8sClient dynamic.Interface) *requestsHandler {
+	return newRequestsHandlerWithClock(logger, k8sClient, &clock.RealClock{})
+}
+
+func (h *requestsHandler) Run(stopCh <-chan struct{}) {
+	go h.runGC(stopCh)
+	<-stopCh
+}
+
+func (h *requestsHandler) CreateRequest(ctx context.Context, request *Request) (string, error) {
+	requestID := uuid.NewString()
+	if err := h.createTraceflow(ctx, requestID, request.Object); err != nil {
+		return "", err
+	}
+	return requestID, nil
+}
+
+func (h *requestsHandler) GetRequestStatus(ctx context.Context, requestID string) (*RequestStatus, error) {
+	_, done, err := h.getTraceflow(ctx, requestID)
+	return &RequestStatus{
+		Done: done,
+		Err:  err,
+	}, nil
+}
+
+func (h *requestsHandler) GetRequestResult(ctx context.Context, requestID string) (map[string]interface{}, error) {
+	object, done, err := h.getTraceflow(ctx, requestID)
+	if err != nil {
+		return nil, err
+	}
+	if !done {
+		return nil, fmt.Errorf("Traceflow not complete yet")
+	}
+	return object, nil
+}
+
+func (h *requestsHandler) DeleteRequest(ctx context.Context, requestID string) (bool, error) {
+	tfName := requestID
+	err := h.k8sClient.Resource(traceflowGVR).Delete(ctx, tfName, metav1.DeleteOptions{})
+	if apierrors.IsNotFound(err) {
+		return false, nil
+	}
+	if err != nil {
+		return false, err
+	}
+	return true, nil
+}
+func (h *requestsHandler) getTraceflow(ctx context.Context, tfName string) (map[string]interface{}, bool, error) {
+	traceflow, err := h.k8sClient.Resource(traceflowGVR).Get(ctx, tfName, metav1.GetOptions{})
+	if err != nil {
+		return nil, false, err
+	}
+	phase, ok, err := unstructured.NestedString(traceflow.Object, "status", "phase")
+	if err != nil {
+		return nil, false, err
+	}
+	if !ok {
+		return traceflow.Object, false, nil
+	}
+	return traceflow.Object, (phase == "Succeeded" || phase == "Failed"), nil
+}
+
+func (h *requestsHandler) createTraceflow(ctx context.Context, tfName string, object map[string]interface{}) error {
+	traceflow := &unstructured.Unstructured{
+		Object: map[string]interface{}{
+			"apiVersion": traceflowGVR.Group + "/" + traceflowGVR.Version,
+			"kind":       "Traceflow",
+			"metadata": map[string]interface{}{
+				"name": tfName,
+				"labels": map[string]interface{}{
+					"ui.antrea.io": "",
+				},
+			},
+			"spec": object["spec"],
+		},
+	}
+	if _, err := h.k8sClient.Resource(traceflowGVR).Create(ctx, traceflow, metav1.CreateOptions{}); err != nil {
+		return err
+	}
+	return nil
+}
+
+func (h *requestsHandler) doGC(ctx context.Context) {
+	labelSelector := metav1.LabelSelector{
+		MatchLabels: map[string]string{"ui.antrea.io": ""},
```

Comment:
Please take a look and tell me if this is what you meant.
Because of the different types expected by the dynamic client (`map[string]interface{}`) and by `metav1.LabelSelector` (`map[string]string`), I had to use the `SetLabels` helper function.